adr/0008-store-key-sanitization.md

```diff
@@ -0,0 +1,89 @@
+# 8: Store Key Sanitization
+
+Date: 2024-07-02
+
+## Status
+
+Under Review
+
+## Context
+
+Issues were discovered when accepting arbitrary Keycloak client IDs and setting them into the Store without sanitization. For instance, IDs like `https://google.com/` failed during Store.setItem due to the `/` character, resulting in a `422` error.
+
+
+## Decision
+
+Implementing base64 would be a heavy shift for an extremely important part of the system. Instead, we will sanitize the key by replacing `/` with a character that does not break the `patch` operation before placing the key. We will also increase the tests to find other edge cases before releasing the change.
+
+#### Sanitize using String Replacement
+* Sanitize the key by replacing `/` with a character that does not break the `patch` operation before placing the key.
```

Comment:
Haven't read too deeply into the specs, but... since the patch operation is a [JSON Patch](https://datatracker.ietf.org/doc/html/rfc6902/#section-4), it's "data" key is defined as a [JSON Pointer](https://datatracker.ietf.org/doc/html/rfc6901#section-3) object... which means that both "~" and "/" are going to have special meanings:

```
A JSON Pointer is a Unicode string (see [[RFC4627], Section 3](https://datatracker.ietf.org/doc/html/rfc4627#section-3))
   containing a sequence of zero or more reference tokens, each prefixed
   by a '/' (%x2F) character.

   Because the characters '~' (%x7E) and '/' (%x2F) have special
   meanings in JSON Pointer, '~' needs to be encoded as '~0' and '/'
   needs to be encoded as '~1' when these characters appear in a
   reference token.
   ```

Reply:
added this link and spoke to the replacement but kept specific implementation as TBD by the testing done during the implementation.